test/rule_test_support.dart

```diff
@@ -49,12 +61,164 @@ class AnalysisOptionsFileConfig {
   }
 }
 
+/// A description of a diagnostic that is expected to be reported.
+class ExpectedDiagnostic {
```

Comment:
Consider splitting this file into several smaller files, maybe in a subdirectory named `support` or `utilities` with each file following the normal file naming conventions.

Reply:
💯 

I'll do this in a follow-up.